homeassistant/components/nanogrid_air/config_flow.py

```diff
@@ -0,0 +1,35 @@
+"""Config flow for Nanogrid Air integration."""
+
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+
+from .api import fetch_mac
+from .const import DOMAIN
+
+API_DEFAULT = "http://ctek-ng-air.local/"
+
+
+class NanogridAirConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Nanogrid Air integration."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize flow."""
+        self._url = API_DEFAULT
+
+    async def async_step_user(self, user_input=None) -> ConfigFlowResult:
+        """Handle a flow initiated by the user."""
+
+        self._async_abort_entries_match()
+
+        mac_address = await fetch_mac()
+        if any(mac_address):
+            unique_id = mac_address
+            await self.async_set_unique_id(unique_id)
+            self._abort_if_unique_id_configured()
+        else:
+            return self.async_abort(reason="not_responding")
+
+        return self.async_create_entry(title="Nanogrid Air", data={"url": self._url})
```

Comment:
Seems like there is no option for the user to manually assign a IP if needed?

Reply:
No we don't want the user to be given that option, the code itself is supposed to do all the work. The less the user need to do, the better, according to HA.

Reply:
What about case where mDNS is not allowed for reasons? Allowing a fallback does not mean we don't provide "single click configuration" as well :)

Reply:
Fixed